app/controllers/mailing_list/steps_controller.rb

```diff
@@ -40,14 +45,24 @@ def crm_store
       session[:mailinglist_crm] ||= {}
     end
 
+    def campaign_id
+      params[:campaign_id]
+    end
+
     def resolve_layout
       is_first_page = @current_step.instance_of? MailingList::Steps::Name
-
+      return "registration_with_campaign" if is_first_page && campaign_id.present?
       return "registration_with_side_images" if is_first_page
 
       "registration"
     end
 
+    def load_campaign_frontmatter
+      campaigns = YAML.load_file(Rails.root.join("config/campaigns.yml"))
```

Comment:
Is there any reason not to pre-load this somewhere instead of doing it for every request? Minor performance issue, but it should be easy to load this at boot time somehow.

Reply:
I thought about this but opted not to; put a note in the PR description -- happy to do so if you disagree, but it felt like a premature optimisation to me and it often makes the tests more complex:

> I've opted not to memoize the campaign data as its currently a very small file and the page appears to load quickly anyway; we can add it in later if we find it necessary.

Reply:
Sorry, missed that note. I think that's fine, normally that file should, hopefully, remain in cache and the only cost would be a few system calls, no big deal. I guess it risks becoming a performance issue if the host is swamped anyway, and can't afford to keep that file cached, which seems rather unlikely.